code/src/hooks/use-redux.ts

```diff
@@ -0,0 +1,6 @@
+import { useDispatch, useSelector, TypedUseSelectorHook } from "react-redux";
+
+import type { RootState, AppDispatch } from "@/store/store";
+
+export const useAppDispatch = () => useDispatch<AppDispatch>();
```

Comment:
[Official docs](https://redux-toolkit.js.org/usage/usage-with-typescript) provide example with `withTypes` method. What do you think about it?

![image](https://github.com/idx-academy/spa-orders/assets/109922101/826bb7b1-4d09-4ce6-b82c-5042918323a1)

Reply:
I agree, fixed